lib/manager/components/transform/NormalizeField.js

```diff
@@ -24,21 +25,71 @@ import type {
 } from '../../../types'
 
 type State = {
-  activeEditingIndex: number
+  activeEditingIndex: number,
+  // Holds the contents that should be shown in the SubstitutionRow component.
+  activeSubstitution: ?Substitution,
+  // Last edited contents.
+  previousEditingIndex: number,
+  previousSubstitution: ?Substitution
+}
+
+/**
+ * Determines whether two substitutions are equal (except for the valid field).
+ */
+function areEqual (sub1: ?Substitution, sub2: ?Substitution): boolean {
```

Comment:
```suggestion
function substitutionsAreEqual (sub1: ?Substitution, sub2: ?Substitution): boolean {
```

Reply:
Updated in https://github.com/ibi-group/datatools-ui/pull/687/commits/cf0d3a024691f9841665e88aba94cd027048044b.